packages/gatsby/src/utils/page-data.js

```diff
@@ -19,7 +21,20 @@ const write = async ({ publicDir }, page, result) => {
     matchPath: page.matchPath,
     result,
   }
-  await fs.outputFile(filePath, JSON.stringify(body))
+  const bodyStr = JSON.stringify(body)
+  // transform asset size to kB (from bytes) to fit 64 bit to numbers
+  const pageDataSize = Buffer.byteLength(bodyStr) / 1000
+
+  const telemetryCache = getCache(`gatsby-telemetry`)
+  const pageDataSizeCache = (await telemetryCache.get(`pageDataStats`)) || {}
+  pageDataSizeCache[filePath] = pageDataSize
+  await telemetryCache.set(`pageDataStats`, pageDataSizeCache)
```

Comment:
Yeah optimally we'd do this only once but I had a hard time finding the right way to do this, since we can't run async code in our exit handlers and `process.exit()` calls seem like a somewhat common cause of the process exiting.

@pieh should we hop on a call at some point to discuss alternatives, say using the redux store?

Reply:
If we choose to use redux for this, we wouldn't need to use exit handler - redux store is configured to periodically persist, so even if there is some page-data stats that weren't persisted, that's ok because persisted state will be consisted, meaning that we will run queries again so we will regenerate query results (and stats) on next run